central/resourcecollection/service/service_impl.go

```diff
@@ -134,6 +153,29 @@ func (s *serviceImpl) CreateCollection(ctx context.Context, request *v1.CreateCo
 	return &v1.CreateCollectionResponse{Collection: collection}, nil
 }
 
+func (s *serviceImpl) UpdateCollection(ctx context.Context, request *v1.UpdateCollectionRequest) (*v1.UpdateCollectionResponse, error) {
+	if !features.ObjectCollections.Enabled() {
+		return nil, errors.New("Resource collections is not enabled")
+	}
+
+	if request.GetId() == "" {
+		return nil, errors.Wrap(errox.InvalidArgs, "Non empty collection id must be specified to delete a collection")
+	}
+
+	collection, err := collectionRequestToCollection(ctx, request, false)
+	if err != nil {
+		return nil, err
+	}
+	collection.Id = request.GetId()
+
+	err = s.datastore.UpdateCollection(ctx, collection)
```

Comment:
You need to fetch the stored collection object and reconcile the updated fields with it i.e. the `createdAt` and `createdBy`. Update the `UpdateCollection` func and add a unit test.

Reply:
Good catch, made that change on `UpdateCollection` implementation on datastore instead as datastore fetches the storedCollection anyways.